src/app-layout/utils/use-drawers.ts

```diff
@@ -74,23 +76,25 @@ function useRuntimeDrawers(
       const localDrawers = drawers.filter(drawer => drawer.type !== 'global');
       const globalDrawers = drawers.filter(drawer => drawer.type === 'global');
       setRuntimeDrawers(convertRuntimeDrawers(localDrawers, globalDrawers));
-      if (!drawersWereOpenRef.current) {
+      if (!localDrawerWasOpenRef.current) {
         const defaultActiveLocalDrawer = sortByPriority(localDrawers).find(drawer => drawer.defaultActive);
         if (defaultActiveLocalDrawer) {
           onLocalDrawerChangeStable(defaultActiveLocalDrawer.id);
         }
-
-        const defaultActiveGlobalDrawers = sortByPriority(globalDrawers).filter(drawer => drawer.defaultActive);
-        defaultActiveGlobalDrawers.forEach(drawer => {
-          onGlobalDrawersChangeStable(drawer.id);
-        });
       }
+
+      const defaultActiveGlobalDrawers = sortByPriority(globalDrawers).filter(
+        drawer => !activeGlobalDrawersIdsStable.current.includes(drawer.id) && drawer.defaultActive
```

Comment:
yes, this way, using the ref. otherwise, if we check `activeGlobalDrawersIds` and add it to the deps array, we fall into infinite loop